### PR TITLE
Fix Python gpu tests

### PR DESCRIPTION
**Pull request by Hardcode84**
* `LowerTakeContextOp`: insert proper type conversion on function call boundaries
* `ConvertAtomicOps`: increase pattern benefit. We are representing gpu atomic ops as calls to functions with some magical names, but upstream recently added theirs own lowering for functions calls in spirv. Increase benefit of our pattern so it is always running first.

### CODE REVIEW

**Comment by fschlimb** [comment]
I am happy to review this but I need a few code-comments to understand the flow and intent.

**Comment by Hardcode84** [comment]
I've added some comments but proper documentation is out of scope of this PR.
And I'll merge this later today anyway as without it our end-to-end python gpu pipeline is broken.

**Comment by fschlimb** [comment]
> And I'll merge this later today anyway as without it our end-to-end python gpu pipeline is broken.

Why are you asking for a review then?

File: mlir/lib/Conversion/gpu_to_gpu_runtime.cpp [hunk 1]
```diff
@@ -829,7 +829,10 @@ static mlir::Value lowerFloatSubAtomic(mlir::OpBuilder &builder,
 
 class ConvertAtomicOps : public mlir::OpConversionPattern<mlir::func::CallOp> {
```
Comment (fschlimb) [inline mlir/lib/Conversion/gpu_to_gpu_runtime.cpp:830]: convert to what?

**Pull request merged**